### PR TITLE
Address inspection hints

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Comment by ben-manes** [comment]
lgtm, thanks! Seems there are some minor KtLint errors, so can merge once the checks pass.

**Comment by sschuberth** [comment]
> lgtm, thanks! Seems there are some minor KtLint errors, so can merge once the checks pass.

Should be addressed now, incl. on more inspection fix.

BTW, very cool that this is Kotlin now 😀

**Review by ben-manes — APPROVED** [review]

**Comment by sschuberth** [comment]
Any idea why [lines like this](https://github.com/ben-manes/gradle-versions-plugin/actions/runs/3919910946/jobs/6701212214#step:5:1285) are gone from the actual output with my first commit in this PR? Is it really correct that these lines are expected to be there?

**Comment by ben-manes** [comment]
That is the `dependency.reason`. Maybe your refactoring to Coordinate no longer provides it?

File: gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt [hunk 1]
```diff
@@ -402,10 +401,10 @@ class Resolver(
 
   private fun getResolvableDependencies(configuration: Configuration): List<Coordinate> {
     val coordinates = configuration.dependencies
-      .filter { dependency -> dependency is ExternalDependency }
-      .map { dependency ->
+      .filterIsInstance<ExternalDependency>()
+      .mapTo(mutableListOf()) { dependency ->
         Coordinate.from(dependency)
-      } as MutableList<Coordinate>
```
Comment (ben-manes) [inline gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt:408]: I would guess that this uses a different factory method now, causing the constraint reason to be lost.

Comment (sschuberth) [reply gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt:408]: I believe it was something different, see the update.

**Review by ben-manes — APPROVED** [review]

**Pull request merged**